src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -899,8 +899,14 @@ MulticopterPositionControl::get_cruising_speed_xy()
 	/*
 	 * in mission the user can choose cruising speed different to default
 	 */
-	return ((PX4_ISFINITE(_pos_sp_triplet.current.cruising_speed) && (_pos_sp_triplet.current.cruising_speed > 0.1f)) ?
-		_pos_sp_triplet.current.cruising_speed : _params.vel_cruise_xy);
+	const float &cruising_speed = _pos_sp_triplet.current.cruising_speed;
+
+	if (PX4_ISFINITE(cruising_speed) && cruising_speed >= 0.0f && cruising_speed <= _params.vel_max_xy) {
```

Comment:
you could have kept the `cond ? if_true : if_false` format here I suppose

Reply:
I'm not a fan of the ternary operator. It usually generates the same code as an if-then-else, but it's harder to read and has a few gotchas due to low operator precedence.